lib/text/vtt_text_parser.js

```diff
@@ -219,6 +222,75 @@ shaka.text.VttTextParser = class {
     return cue;
   }
 
+  /**
+   * Parses a WebVTT styles from the given payload.
+   *
+   * @param {string} payload
+   * @param {!shaka.text.Cue} rootCue
+   */
+  static parseCueStyles(payload, rootCue) {
+    const xmlPayload = '<span>' + payload + '</span>';
+    const element = shaka.util.XmlUtils.parseXmlString(xmlPayload, 'span');
+    if (element) {
+      /** @type {Array.<!shaka.extern.Cue>} */
+      const cues = [];
+      const VttTextParser = shaka.text.VttTextParser;
+      const childNodes = [...element.childNodes];
+      if (childNodes.length == 1) {
+        const childNode = childNodes[0];
+        if (childNode.nodeType == Node.TEXT_NODE ||
+            childNode.nodeType == Node.CDATA_SECTION_NODE) {
+          rootCue.payload = payload;
+          return;
+        }
+      }
+      for (const childElement of childNodes) {
+        VttTextParser.generateCueFromElement_(childElement, rootCue, cues);
+      }
+      rootCue.nestedCues = cues;
+    } else {
+      rootCue.payload = payload;
```

Comment:
This case only occurs with a parser error, and that shouldn't happen for well-formed cues.  So I would suggest logging a warning that the cue's markup could not be parsed.

Reply:
I added a log with the warning